src/riot-rs-threads/src/scheduler.rs

```diff
@@ -0,0 +1,58 @@
+use critical_section::CriticalSection;
```

Comment:
nit: filename doesn't fit IMO, "the scheduler" is either the whole crate, or just `RunQueue`.